zun/container/k8s/driver.py

```diff
@@ -850,6 +855,27 @@ def inspect_network(self, network):
     #
 
     def pull_image(self, context, repo, tag, image_pull_policy, image_driver_name, **kwargs):
+        for secret_info in self.get_secrets_for_image(repo, context):
+            # Create a new secret for an existing registry
+            if secret_info["registry"] and not secret_info["secret"]:
+                LOG.info(f"Creating new secret {secret_info['name']}")
+                secret = client.V1Secret()
+                secret.metadata = client.V1ObjectMeta(name=secret_info["name"])
+                secret.type = "kubernetes.io/dockerconfigjson"
+                username = secret_info['registry'].username
+                password = secret_info['registry'].password
+                auth = base64.b64encode(f"{username}:{password}".encode("utf-8")).decode("utf-8")
+                data = {
+                    "auths": {
+                        secret_info["registry"].domain: {
```

Comment:
Interesting, so all that is needed is the `auth` encoded parameter? I thought username/password was also on there typically.

Reply:
that's correct, you can use either a username/password, or an auth token. K8s docs have some details here:
https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials